src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/SyncAndMigrateResourceTest.kt

```diff
@@ -270,13 +269,131 @@ class SyncAndMigrateResourceTest : SqsIntegrationTestBase() {
         .isBadRequest
     }
 
+    @Test
+    fun `can sync non-association by ID`() {
+      val existingNa = repository.save(
+        genNonAssociation(
+          firstPrisonerNumber = "A7777XT",
+          secondPrisonerNumber = "B7777XT",
+        ),
+      )
+      val request = UpsertSyncRequest(
+        id = existingNa.id,
+        firstPrisonerNumber = "DUMMY",
+        firstPrisonerReason = LegacyReason.VIC,
+        secondPrisonerNumber = "DUMMY",
+        secondPrisonerReason = LegacyReason.PER,
+        restrictionType = LegacyRestrictionType.CELL,
+        effectiveFromDate = LocalDate.now(clock).minusDays(10),
+      )
+
+      val expectedResponse =
+        // language=json
+        """
+        {
+          "firstPrisonerNumber": "${existingNa.firstPrisonerNumber}",
+          "firstPrisonerRole": "VICTIM",
+          "firstPrisonerRoleDescription": "Victim",
+          "secondPrisonerNumber": "${existingNa.secondPrisonerNumber}",
+          "secondPrisonerRole": "PERPETRATOR",
+          "secondPrisonerRoleDescription": "Perpetrator",
+          "reason": "OTHER",
+          "reasonDescription": "Other",
+          "restrictionType": "${request.restrictionType.toRestrictionType()}",
+          "restrictionTypeDescription": "${request.restrictionType.toRestrictionType().description}",
+          "comment": "$NO_COMMENT_PROVIDED",
+          "authorisedBy": "",
```

Comment:
do you think `authorisedBy` should be empty when the sync request provides a null? or should we maybe use `SYSTEM_USERNAME`…

Reply:
so we could have

```kotlin
authorisedBy = syncRequest.authorisedBy
```

like this

```kotlin
closedBy = syncRequest.authorisedBy ?: SYSTEM_USERNAME
```

a few lines below it

Reply:
yes probably!